surveyor.py

```diff
@@ -5,477 +5,457 @@
-                _write_results(writer, nested_results, tag.tag, str(tag.data), tag, log)
-
-        if output_file:
-            log_echo(f"\033[95mResults saved: {output_file.name}\033[0m", log)
-    except KeyboardInterrupt:
-        log_echo("Caught CTRL-C. Exiting...", log)
-    except Exception as e:
-        log_echo(f'Caught {type(e).__name__} (see log for details): {e}', log, logging.ERROR)
-        log.exception(e)
-    finally:
-        if output_file:
-            output_file.close()
-
-
-def create_generic_product_command(name: str) -> Callable:
-    @click.pass_context
-    def command(ctx):
-        survey(ctx, name)
-
-    command.__name__ = name
-    return command
-
-
-# create click commands for all products that don't have a command function defined
-for product_name in get_products():
-    dir_res = dir()
-    if product_name not in dir_res:
-        cli.command(name=product_name, help=f'Query {product_name}')(create_generic_product_command(str(product_name)))
+                print(self._table_template_str.format(*row))
 
 if __name__ == "__main__":
-    cli()
+    
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--prefix", help="Output filename prefix.", type=str)
+    parser.add_argument("--profile", help="The credentials profile to use.", type=str)
+    parser.add_argument("--days", help="Number of days to search.", type=int)
+    parser.add_argument("--minutes", help="Number of minutes to search.", type=int)
+    parser.add_argument("--limit",help="""
+                Number of results to return. Cortex XDR: Default: 1000, Max: Default
+                Microsoft Defender for Endpoint: Default/Max: 100000
+                SentinelOne (PowerQuery): Default/Max: 1000
+                SentinelOne (Deep Visibility): Default/Max: 20000
+                VMware Carbon Black EDR: Default/Max: None
+                VMware Carbon Black Cloud Enterprise EDR: Default/Max: None
+                
+                Note: Exceeding the maximum limits will automatically set the limit to its maximum value, where applicable.
+                """
+                , type=int)
+
+    parser.add_argument("--hostname", help="Target specific host by name.", type=str)
+    parser.add_argument("--username", help="Target specific username.",  type=str)
+
+    # different ways you can survey the EDR
+    parser.add_argument("--deffile", help="Definition file to process (must end in .json).", type=str)
+    parser.add_argument("--defdir", help="Directory containing multiple definition files.", type=str)
+    parser.add_argument("--query", help="A single query to execute.", type=str)
+    parser.add_argument("--iocfile", help="IOC file to process. One IOC per line. REQUIRES --ioctype")
+    parser.add_argument("--ioctype", help="One of: ipaddr, domain, md5", choices=['ipaddr', 'domain', 'md5'])
+    parser.add_argument("--iocsource", help="One of: ipaddr, domain, md5", type=str, default= None)
```

Comment:
bad documentation here, they are different

- `ioc_source` (str, optional): The source of the IOCs specified in ioc_list.
- `ioc_type` (str, optional): The type or category of IOCs to search for during the survey.

These two are additional arguments to increase flexibility of Surveyor. (assuming additional FR's that'll eventually look to enable this)

Also going back to add the option to pass in a file for ioc searches as well if calling surveyor in-script, since the functionality is available with Sigma, and Definition Files.